src/Interfaces/MOI_interface.jl

```diff
@@ -874,25 +872,23 @@ const status_dual_dict = Dict(
     SOLVED_TO_ACCEPTABLE_LEVEL => MOI.NEARLY_FEASIBLE_POINT,
     INFEASIBLE_PROBLEM_DETECTED => MOI.INFEASIBLE_POINT)
 
-termination_status(nlp::NonlinearProgram) = haskey(status_moi_dict,nlp.status) ?
-    status_moi_dict[nlp.status] : MOI.UNKNOWN_RESULT_STATUS
-termination_status(ips::Solver) = termination_status(ips.nlp)
-primal_status(ips::Solver) = haskey(status_primal_dict,ips.nlp.status) ?
-    status_primal_dict[ips.nlp.status] : MOI.UNKNOWN_RESULT_STATUS
-dual_status(ips::Solver) = haskey(status_dual_dict,ips.nlp.status) ?
-    status_dual_dict[ips.nlp.status] : MOI.UNKNOWN_RESULT_STATUS
+termination_status(ips::Solver) = haskey(status_moi_dict,ips.status) ? status_moi_dict[ips.status] : MOI.UNKNOWN_RESULT_STATUS
+primal_status(ips::Solver) = haskey(status_primal_dict,ips.status) ?
+    status_primal_dict[ips.status] : MOI.UNKNOWN_RESULT_STATUS
+dual_status(ips::Solver) = haskey(status_dual_dict,ips.status) ?
+    status_dual_dict[ips.status] : MOI.UNKNOWN_RESULT_STATUS
 
 
 function MOI.get(model::Optimizer, attr::MOI.ObjectiveValue)
     MOI.check_result_index_bounds(model, attr)
     scale = (model.sense == MOI.MAX_SENSE) ? -1 : 1
-    return scale * model.nlp.obj_val
+    return scale * model.ips.obj_val
 end
 
 function MOI.get(model::Optimizer, attr::MOI.VariablePrimal, vi::MOI.VariableIndex)
     MOI.check_result_index_bounds(model, attr)
     check_inbounds(model, vi)
-    return model.nlp.x[vi.value]
+    return model.ips.x[vi.value]
```

Comment:
I think this is a net improvement: instead of having two primal variables `ips.x` and `ips.nlp.x`, now everything is stored inside `ips`

Reply:
now we have a separate result in `AbstractExecutionStats` of `SolverCore.jl`, but we don't allocate memory (just return it as a view).